Examples/Sources/MacroExamples/Implementation/Extension/SendableExtensionMacro.swift

```diff
@@ -0,0 +1,39 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxMacros
+
+public enum SendableExtensionMacro: ExtensionMacro {
+  public static func expansion(
+    of node: AttributeSyntax,
+    attachedTo declaration: some DeclGroupSyntax,
+    providingExtensionsOf type: some TypeSyntaxProtocol,
+    conformingTo protocols: [TypeSyntax],
+    in context: some MacroExpansionContext
+  ) throws -> [ExtensionDeclSyntax] {
+    if protocols.isEmpty {
+      return []
+    }
+
+    let sendableExtension: DeclSyntax =
+      """
+      extension \(type.trimmed): Sendable {}
+      """
+
+    guard let extensionDecl = sendableExtension.as(ExtensionDeclSyntax.self) else {
+      return []
+    }
```

Comment:
@ahoppen do we have any guidelines on the canonical way to do this? Seems like:
```
let sendableExtension = try ExtensionDeclSyntax("extension \(type.trimmed): Sendable {}")
```
would be nicer to me, since we don't expect that to ever actually fail. We could also actually use `protocols` here rather than assuming `Sendable`.

Reply:
I'd suggest sticking with `let sendableExtension = try ExtensionDeclSyntax("extension \(type.trimmed): Sendable {}")` for now. We currently lack the capability to pass `protocols` in our unit tests, so keeping this implementation will enable us to write tests.

Reply:
> We currently lack the capability to pass protocols in our unit tests, so keeping this implementation will enable us to write tests.

Indeed , though that's something else we need to fix :)

Reply:
Done. Tests are passing ✅

Reply:
Thanks! 🥳

Reply:
The pleasure is all mine 😍